tests/componenttests/server/tests/mediaPipelineCapabilities/MediaPipelineCapabilitiesTest.cpp

```diff
@@ -37,23 +46,32 @@ class MediaPipelineCapabilitiesTest : public RialtoServerComponentTest
         willConfigureSocket();
         configureSutInActiveState();
         connectClient();
+
+        m_dummyParams[0].name = kPropertyName1;
+        m_dummyParams[1].name = kPropertyName3;
+        m_dummyParams[2].name = kPropertyName2;
+        for (int i = 0; i < kNumPropertiesOnSink; ++i)
+            m_dummyParamsPtr[i] = &m_dummyParams[i];
     }
     ~MediaPipelineCapabilitiesTest() override = default;
 
     void willCallGetSupportedProperties()
     {
-        const GType kDummyType{3};
         GstElementFactory *dummyFactory{nullptr};
-        memset(&m_dummyClass, 0x00, sizeof(m_dummyClass));
         GstElementFactoryListType expectedFactoryListType{
             GST_ELEMENT_FACTORY_TYPE_SINK | GST_ELEMENT_FACTORY_TYPE_DECODER | GST_ELEMENT_FACTORY_TYPE_MEDIA_VIDEO};
         m_listOfFactories = g_list_append(m_listOfFactories, dummyFactory);
         EXPECT_CALL(*m_gstWrapperMock, gstElementFactoryListGetElements(expectedFactoryListType, GST_RANK_NONE))
             .WillOnce(Return(m_listOfFactories));
-        EXPECT_CALL(*m_gstWrapperMock, gstElementFactoryGetElementType(dummyFactory)).WillOnce(Return(kDummyType));
-        EXPECT_CALL(*m_glibWrapperMock, gTypeClassRef(kDummyType)).WillOnce(Return(&m_dummyClass));
-        EXPECT_CALL(*m_glibWrapperMock, gObjectClassFindProperty(&m_dummyClass, _)).WillRepeatedly(Return(&m_dummyParam));
-        EXPECT_CALL(*m_glibWrapperMock, gObjectUnref(&m_dummyClass));
+        // The next calls should ensure that an object is created and then freed
+        GstElement object;
```

Comment:
Variables here should be in the test class/global, EXPECT_CALLs are initialised here but aren't actually 'used' until the mock is called in the test. When the mock is called, these objects will be out of scope which can cause problems with the test.

Reply:
Actually, I'm wrong... G_OBJECT_GET_CLASS() is called on &object and therefore it's an excellent point that you're making! (And that's why I had to do the memset in the above code). I'm moving it...